validation/promotion_test.go

```diff
@@ -13,12 +13,15 @@ func TestPromotion(t *testing.T) {
 		t.Run(perChainTestName(chain), func(t *testing.T) {
```

Comment:
```suggestion
// WARNING: this test must not run along side any other tests, because it mutates some global object.
// It should be strictly isolated. 
		t.Run(perChainTestName(chain), func(t *testing.T) {
```